master/internal/api_command.go

```diff
@@ -27,3 +40,52 @@ func (a *apiServer) KillCommand(
 	_ context.Context, req *apiv1.KillCommandRequest) (resp *apiv1.KillCommandResponse, err error) {
 	return resp, a.actorRequest(fmt.Sprintf("/commands/%s", req.CommandId), req, &resp)
 }
+
+func (a *apiServer) LaunchCommand(
+	ctx context.Context, req *apiv1.LaunchCommandRequest,
+) (*apiv1.LaunchCommandResponse, error) {
+
+	user, _, err := grpc.GetUser(ctx, a.m.db)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get the user: %s", err)
+	}
+
+	cmdParams := command.CommandParams{UserFiles: filesToArchive(req.Files)}
+	if req.TemplateName != "" {
+		cmdParams.Template = &req.TemplateName
+	}
+	if req.Config != nil {
+		configBytes, err := protojson.Marshal(req.Config)
+		if err != nil {
+			return nil, err
+		}
+		cmdParams.ConfigBytes = configBytes
+	}
+	if len(req.Data) != 0 {
+		var data map[string]interface{}
+		if err := json.Unmarshal(req.Data, &data); err != nil {
+			return nil, err
+		}
+		cmdParams.Data = data
+	}
```

Comment:
Re: ci complaining about dupes, for now pull this out into a helper method and pass the individual, shared, inner portions of the LaunchRequests?  Something like command.NewCommandParams() that takes a union of all fields different launch requests provide.

Reply:
okay I pushed some final changes. I didn't push the helper into the command package to keep proto definitions closer to the API and not into the command package :thinking: